pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -125,20 +132,27 @@ func NewVolumeBinder(
 	bindTimeout time.Duration) SchedulerVolumeBinder {
 
 	b := &volumeBinder{
-		kubeClient:      kubeClient,
-		classLister:     storageClassInformer.Lister(),
-		nodeInformer:    nodeInformer,
-		pvcCache:        NewPVCAssumeCache(pvcInformer.Informer()),
-		pvCache:         NewPVAssumeCache(pvInformer.Informer()),
-		podBindingCache: NewPodBindingCache(),
-		bindTimeout:     bindTimeout,
+		kubeClient:       kubeClient,
+		classLister:      storageClassInformer.Lister(),
+		nodeInformer:     nodeInformer,
+		pvcCache:         NewPVCAssumeCache(pvcInformer.Informer()),
+		pvCache:          NewPVAssumeCache(pvInformer.Informer()),
+		podBindingCache:  NewPodBindingCache(),
+		podToAssumedNode: map[string]string{},
+		bindTimeout:      bindTimeout,
 	}
 
 	return b
 }
 
-func (b *volumeBinder) GetBindingsCache() PodBindingCache {
-	return b.podBindingCache
+func (b *volumeBinder) DeletePodBindings(pod *v1.Pod) {
+	podName := getPodName(pod)
+	if nodeName, ok := b.podToAssumedNode[podName]; ok {
+		b.revertAssumedPVs(b.podBindingCache.GetBindings(pod, nodeName))
+		b.revertAssumedPVCs(b.podBindingCache.GetProvisionedPVCs(pod, nodeName))
+		delete(b.podToAssumedNode, podName)
+	}
+	b.podBindingCache.DeleteBindings(pod)
```

Comment:
`podToAssumedNode` is unnecessary, you can use `PodBindingCache.GetDecisions(pod *v1.Pod)` to retrieve all decisions then revert related assumed PV/PVC cache.

Reply:
I think `PodBindingCache` doesn't know which node was selected. `podBindingCache.GetDecisions(pod *v1.Pod)` returns decisions for each node. It is not correct to restore cache of not assumed by the target pod. That cache might have been changed by another pod.